apps/nightscout/nightscout.star

```diff
@@ -0,0 +1,210 @@
+
+def get_schema():
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Text(
+                id = "nightscout_id",
+                name = "Nightscout ID",
+                desc = "Your Nightscout ID (i.e. XXX in https://XXX.herokuapp.com/api/v1/entries.json)",
+                icon = "user"
+            ),
+            schema.Text(
+                id = "normal_high",
+                name = "Normal High Threshold",
+                desc = "Anything above this is displayed yellow unless it is above the Urgent High Threshold (default " + str(DEFAULT_NORMAL_HIGH) + ")",
+                icon = "inputNumeric"
+            ),
+            schema.Text(
+                id = "normal_low",
+                name = "Normal Low Threshold",
+                desc = "Anything below this is displayed yellow unless it is below the Urgent Low Threshold (default " + str(DEFAULT_NORMAL_LOW) + ")",
+                icon = "inputNumeric"
+            ),
+            schema.Text(
+                id = "urgent_high",
+                name = "Urgent High Threshold",
+                desc = "Anything above this is displayed red (Default " + str(DEFAULT_URGENT_HIGH) + ")",
+                icon = "inputNumeric"
+            ),
+            schema.Text(
+                id = "urgent_low",
+                name = "Urgent Low Threshold",
+                desc = "Anything below this is displayed red (Default " + str(DEFAULT_URGENT_LOW) + ")",
+                icon = "inputNumeric"
+            )   
+        ]
+    )
+
+def main(config):
+    nightscout_id = config.get("nightscout_id", None)
+    normal_high = int(config.get("normal_high", DEFAULT_NORMAL_HIGH))
+    normal_low = int(config.get("normal_low", DEFAULT_NORMAL_LOW))
+    urgent_high = int(config.get("urgent_high", DEFAULT_URGENT_HIGH))
+    urgent_low = int(config.get("urgent_low", DEFAULT_URGENT_LOW))
+
+    if nightscout_id == None:
+        print("No Nightscout ID provided - can't do anything")
+        return render.Root(
+            render.Marquee(
+                width = 64,
+                child = render.Text(
+                    content=str("Please specify your Nightscout ID")
+                )
+            )
+        )
+
+    # Get the JSON object from the cache
+    nightscout_data_cached = cache.get(nightscout_id + "_nightscout_data")
+    if nightscout_data_cached != None:
+        nightscout_data_json = json.decode(nightscout_data_cached)
+        print("Hit - displaying cached data")
+
+        # Pull the data from the cache
+        sgv_current = int(nightscout_data_json["sgv_current"])
+        sgv_delta = int(nightscout_data_json["sgv_delta"])
+        latest_reading_dt = time.parse_time(nightscout_data_json["latest_reading_date_string"])
+        trend = nightscout_data_json["trend"]
+    else:
+        print("Miss - calling Nightscout API")
+        nightscout_url = "https://" + nightscout_id + ".herokuapp.com/api/v1/entries.json"
```

Comment:
This url created from user input makes me pretty nervous. We really want to avoid the case where someone could make calls to another service that we don't intend or even worse - make calls to our internal systems. Can we add validation to ensure this is actually a valid ID for heroku? We could either do it through inspection of the string or we could potentially use a heroku API if they offer it?